tests/test_utils.py

```diff
@@ -190,19 +190,152 @@ class TestRenameColumns:
 
     def test_rename_columns_success(self):
         renamed_df = utils.rename_columns(
-            df=self.df.copy(), column_map=self.good_column_map
+            data=self.df.copy(), column_map=self.good_column_map
         )
         assert list(renamed_df.columns) == list(self.good_column_map.values())
 
     def test_rename_columns_TypeError(self):
         captured_output = StringIO()
         sys.stdout = captured_output
         bad_renamed_df = utils.rename_columns(
-            df=self.df.copy(), column_map=self.bad_column_map
+            data=self.df.copy(), column_map=self.bad_column_map
         )
         assert "Column mapping must be a dictionary" in captured_output.getvalue()
         assert list(bad_renamed_df.columns) == list(self.good_column_map.keys())
 
+    def test_rename_columns_with_dict_input(self):
+        """Test renaming columns in a dictionary input"""
+        input_dict = {"a": "value1", "b": "value2", "c": "value3"}
+        column_map = {"a": "x", "b": "y", "c": "z"}
+
+        result = utils.rename_columns(data=input_dict, column_map=column_map)
+
+        expected = {"x": "value1", "y": "value2", "z": "value3"}
+        assert result == expected
+        # Verify the original dict was modified in place
+        assert input_dict == expected
```

Comment:
Interesting... if things are modified in place, can you add that to the docstring for `rename_columns` function? It's not clear that that's the behavior for dicts from the code.